core/monitoring/power_monitor.py

```
@@ -0,0 +1,417 @@
+    
+    def _detect_ups(self) -> bool:
+        """Detect if UPS monitoring is available."""
+        ups_tools = [
+            ("apcaccess", "apcupsd"),  # APC UPS
+            ("upsc", "nut"),            # Network UPS Tools
+            ("pwrstat", "cyberpower")   # CyberPower
+        ]
+        
+        for cmd, ups_type in ups_tools:
+            try:
+                result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call with `shell=True` identified, security issue

(S602)